cli/install_ee/install_ee.go

```diff
@@ -112,156 +46,116 @@ func getVersions(data *[]byte) ([]version.Version, error) {
 
 	switch osType {
 	case util.OsLinux:
-		// Bundles without specifying the architecture are all x86_64.
+		// Regexp for bundles from the new layout.
+		matchReNew = "((.+/)(tarantool-enterprise-sdk-(.*r[0-9]{3})?(?:.linux.)?" +
+			arch + "\\.tar\\.gz))"
+
+		// Regexp for bundles from the old layout.
 		if arch == "x86_64" {
-			matchRe = ".*>tarantool-enterprise-bundle-" +
-				"(.*-g[a-f0-9]+-r[0-9]{3}(?:-nogc64)?)(?:-linux-x86_64)?\\.tar\\.gz<.*"
+			matchReOld = "((.+/)(tarantool-enterprise-bundle-" +
+				"(.*-g[a-f0-9]+-r[0-9]{3}(?:-nogc64)?)(?:-linux-x86_64)?\\.tar\\.gz))"
 		} else {
-			matchRe = ".*>tarantool-enterprise-bundle-" +
-				"(.*-g[a-f0-9]+-r[0-9]{3}(?:-nogc64)?)(?:-linux-" + arch + ")\\.tar\\.gz<.*"
+			matchReOld = "((.+/)(tarantool-enterprise-bundle-" +
+				"(.*-g[a-f0-9]+-r[0-9]{3}(?:-nogc64)?)(?:-linux-" + arch + ")\\.tar\\.gz))"
 		}
 	case util.OsMacos:
-		// Bundles without specifying the architecture are all x86_64.
+		matchReNew = "((.+/)(tarantool-enterprise-sdk-(.*r[0-9]{3})?(?:.macos.)?" +
+			arch + "\\.tar\\.gz))"
+
 		if arch == "x86_64" {
-			matchRe = ".*>tarantool-enterprise-bundle-" +
-				"(.*-g[a-f0-9]+-r[0-9]{3})-macos(?:x-x86_64)?\\.tar\\.gz<.*"
+			matchReOld = "((.+/)(tarantool-enterprise-bundle-" +
+				"(.*-g[a-f0-9]+-r[0-9]{3})-macos(?:x-x86_64)?\\.tar\\.gz))"
 		} else {
-			matchRe = ".*>tarantool-enterprise-bundle-" +
-				"(.*-g[a-f0-9]+-r[0-9]{3})(?:-macosx-" + arch + ")\\.tar\\.gz<.*"
+			matchReOld = "((.+/)(tarantool-enterprise-bundle-" +
+				"(.*-g[a-f0-9]+-r[0-9]{3})(?:-macosx-" + arch + ")\\.tar\\.gz))"
 		}
 	}
 
-	re := regexp.MustCompile(matchRe)
-	parsedData := re.FindAllStringSubmatch(strings.TrimSpace(string(*data)), -1)
+	reNew := regexp.MustCompile(matchReNew)
+	reOld := regexp.MustCompile(matchReOld)
+
+	versionRegexpSl[0] = reNew
+	versionRegexpSl[1] = reOld
+
+	return versionRegexpSl, nil
+}
+
+// getVersions collects a list of all available tarantool-ee
+// versions for the host architecture.
+func getVersions(data *[]byte) ([]EEVersion, error) {
+	versions := []EEVersion{}
+
+	regexpSl, err := compileVersionRegexp()
+	if err != nil {
+		return nil, err
+	}
+
+	// Concatenating of new and old parsed versions names.
+	parsedData := append(regexpSl[0].FindAllStringSubmatch(strings.TrimSpace(string(*data)), -1),
+		regexpSl[1].FindAllStringSubmatch(strings.TrimSpace(string(*data)), -1)...)
 	if len(parsedData) == 0 {
 		return nil, fmt.Errorf("no packages for this OS")
 	}
 
 	for _, entry := range parsedData {
-		version, err := version.GetVersionDetails(entry[1])
+		eeVersion, err := GetEEVersionDetails(entry[4])
 		if err != nil {
 			return nil, err
 		}
-		version.Tarball, err = getTarballName(entry[0])
-		if err != nil {
-			return nil, err
-		}
-		versions = append(versions, version)
+		eeVersion.VersionInfo.Tarball = entry[3]
+		eeVersion.Prefix = entry[2]
+		versions = append(versions, eeVersion)
 	}
 
-	version.SortVersions(versions)
+	SortEEVersions(versions)
```

Comment:
Now logic is another. I use the BundleInfoSlice type with specific methods for using it in sort function. But I still can not avoid the repeat of `Version`'s comparation logic, due to the static typing or impossibility to cast slices(without using unsafe actions).

Reply:
We need to extract comparison logic to separate function, which accepts two versions. Sorting functions must use it for comparison. SortEE must use custom comparator, which calls version comparator. If I didn't miss anything...

Reply:
The `Less()` method, if I understood you correctly, must accept only two integers to be used in `sort.Sort()` function, it can not accept two variables of the same type AFAIU.

Reply:
Fixed.